pkg/client/kubernetes/manifestoptions.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+// DeleteManifestOption is some configuration that modifies options for a delete request.
+type DeleteManifestOption interface {
+	// MutateDeleteOptions applies this configuration to the given delete options.
+	MutateDeleteManifestOptions(opts *DeleteManifestOptions)
+}
+
+// DeleteOptions contains options for delete requests
+type DeleteManifestOptions struct {
+	// Ignores resources with groups, versions or kinds not registered in the apiserver.
+	IgnoreNoMatch bool
```

Comment:
I suggest to align this with the error toleration func concept we already have in place https://github.com/gardener/gardener/blob/d2cd7c5169b3ec61321959ffa588017539096a9b/pkg/utils/kubernetes/client/options.go#L36.